### PR TITLE
ROX-13433: Adding Extraneous Flows graph

### PR DESCRIPTION
**Pull request by alwayshooin**
This will need to be rebased off of Saif's data structure PR, but it adds the extraneous flows graph logic, refactors some of the node/grouped node generating utils, and adds a new `fakeGroup` type to accommodate for the allowed in/egress flows node that appears for the extraneous flows graph when a node is selected. 

The `ingress flows` and `egress flows` will be added in a separate PR to incrementally add changes

image

active traffic w central selected: 
Screenshot 2022-12-05 at 4 10 51 PM

extraneous flows w central selected:
Screenshot 2022-12-05 at 4 11 30 PM

### CODE REVIEW

**Comment by sachaudh** [comment]
@alwayshooin you can rebase now

**Review by sachaudh — COMMENTED** [review]
LGTM to me so far. I'll wait for approval until the rebase occurs and I take another look

File: ui/apps/platform/src/Containers/NetworkGraph/utils/modelUtils.tsx
Comment (alwayshooin) [reply ui/apps/platform/src/Containers/NetworkGraph/utils/modelUtils.tsx:244]: yeah, i need to figure out how to represent it visually in the graph and add the edges. the ones here are mostly just placeholders for now

File: ui/apps/platform/src/Containers/NetworkGraph/types/topology.type.ts
Comment (sachaudh) [inline ui/apps/platform/src/Containers/NetworkGraph/types/topology.type.ts:14]: What is the `PolicyNodeModel` and how is it different from a `DeploymentNodeModel`? I was wondering more why have a separate type. If a `PolicyNodeModel` has `nonIsolatedIngress` and `nonIsolatedEgress` as false, then would it be the same as a `DeploymentNodeModel`?

**Review by sachaudh — APPROVED** [review]
LGTM. Just a comment

File: ui/apps/platform/src/Containers/NetworkGraph/NetworkGraphPage.tsx
Comment (vjwilson) [inline ui/apps/platform/src/Containers/NetworkGraph/NetworkGraphPage.tsx:52]: Ok for this PR, but I think we need a comment and a Jira task to update this state to be URL-driven, based on the plan in our design doc.

> For controls that cause the front-end to display the data differently, without having to make another API call, we will have to create appropriate param names. Because our Search options are always Title-Case strings with spaces between distinct words, to avoid collisions we should use appropriate camelCase query param names for display-only features, like showConnections=allowed and namespaceFlows=hide
https://docs.google.com/document/d/1QMI0PQ_duKJ_6aqD3SU8gNjBTUDPyNBMOktEG5opABc/edit#heading=h.pan7jow300u7

File: ui/apps/platform/src/Containers/NetworkGraph/NetworkGraphPage.tsx
Comment (vjwilson) [inline ui/apps/platform/src/Containers/NetworkGraph/NetworkGraphPage.tsx:100]: Nit: replace hard-coded position "magic number" with a constant.

```
const POLICY_RESPONSE_INDEX = 1;
...
    const { nodes: policyNodes } = values[POLICY_RESPONSE_INDEX].response;
```

File: ui/apps/platform/src/Containers/NetworkGraph/NetworkGraphPage.tsx
Comment (vjwilson) [inline ui/apps/platform/src/Containers/NetworkGraph/NetworkGraphPage.tsx:113]: Nit: magic number

File: ui/apps/platform/src/Containers/NetworkGraph/components/stylesComponentFactory.tsx
Comment (vjwilson) [inline ui/apps/platform/src/Containers/NetworkGraph/components/stylesComponentFactory.tsx:37]: It looks like we could group this case with `case 'node':` above.

**Review by vjwilson — COMMENTED** [review]
I know this is already merged, but I left a few comments for future considertation.

I noticed a couple of odd behaviors in the `EdgeStateSelect` component when smoke-testing this on a cluster.

1. The dropdown doesn't close on selecting an option.
2. Clicking either option toggles between the two options, which is counter-intuitive. I would expect that clicking an option would be "idempotent".

**Pull request merged**
